partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/TypeEnv.kt

```diff
@@ -29,6 +29,7 @@ internal data class TypeEnv(
     internal fun getScope(depth: Int): TypeEnv {
         return when (depth) {
             0 -> this
+            // TODO: Consider resolve the variable to the correct index
```

Comment:
We calculate the correct index in `Compiler`. There are a couple reasons that we can discuss.